pvHelpers/email_factory.py

```diff
@@ -151,12 +156,18 @@ def fromServerMessage(for_user_id, decrypted_msg, wrapped_key, key_version, mail
             ccs = map(lambda r: EmailHelpers.format_recip(r),
                       decrypted_msg["private_metadata"].get("ccs", []))
 
-            # if sender, we can see all bccs
+            # if sender or gateway user, we can see all bccs;
             # else figure out whether we are bcced.
             lfor_user_id = for_user_id.lower()
-            if lfor_user_id == decrypted_msg["private_metadata"]["sender"].lower():
-                bccs = map(lambda u: {"user_id": u["user_id"], "display_name": u["user_id"]},
+            if lfor_user_id == decrypted_msg["private_metadata"]["sender"].lower() or (common_props.get("other_headers") and common_props["other_headers"].get("X-External-BCCs") and lfor_user_id == common_props["other_headers"]["X-External-BCCs"][0]["user_id"].lower()):
```

Comment:
This condition is hard to read. Could you pull out the external check into its own line?
```python
something = ...
if something
```

Also, is it possible to have multiple external bccs? In this case?

Reply:
Cool, extracted out.
And it definitely is possible to have multiple external bccs - checking only for the first one here just to identify what the gateway user's id is for the org, and then checking that against others. But if I missed what you noticed here please let me know!